tests/e2e/runtime.go

```diff
@@ -0,0 +1,465 @@
+//	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Signer does not implement "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".Signer (wrong type for ContextSign method)
+//		have ContextSign("github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Context, []byte) ([]byte, error)
+//		want ContextSign([]byte, []byte) ([]byte, error)
+type signerWrapper struct {
+	signer coreSig.Signer
+}
+
+func newSignerWrapper(signer coreSig.Signer) *signerWrapper {
+	return &signerWrapper{signer: signer}
+}
+
+func (sw *signerWrapper) Public() signature.PublicKey {
+	return newPublicKeyWrapper(sw.signer.Public())
+}
+
+func (sw *signerWrapper) ContextSign(context, message []byte) ([]byte, error) {
+	return sw.signer.ContextSign(coreSig.Context(context), message)
+}
+
+func (sw *signerWrapper) String() string {
+	return sw.signer.String()
+}
+
+func (sw *signerWrapper) Reset() {
+	sw.signer.Reset()
+}
+
+type kvKey struct {
+	Key string `json:"key"`
+}
+
+type kvKeyValue struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+func kvInsert(rtc client.RuntimeClient, signer signature.Signer, nonce uint64, key string, value string) error {
+	ctx := context.Background()
+
+	tx := types.NewTransaction(nil, "Insert", kvKeyValue{
+		Key:   key,
+		Value: value,
+	})
+	tx.AppendSignerInfo(signer.Public(), nonce)
+	stx := tx.PrepareForSigning()
+	stx.AppendSign(signer)
+
+	if _, err := rtc.SubmitTx(ctx, stx.UnverifiedTransaction()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kvRemove(rtc client.RuntimeClient, signer signature.Signer, nonce uint64, key string) error {
+	ctx := context.Background()
+
+	tx := types.NewTransaction(nil, "Remove", kvKey{
+		Key: key,
+	})
+	tx.AppendSignerInfo(signer.Public(), nonce)
+	stx := tx.PrepareForSigning()
+	stx.AppendSign(signer)
+
+	if _, err := rtc.SubmitTx(ctx, stx.UnverifiedTransaction()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func kvGet(rtc client.RuntimeClient, key string) (string, error) {
+	ctx := context.Background()
+
+	var resp kvKeyValue
+	if err := rtc.Query(ctx, client.RoundLatest, "Get", kvKey{Key: key}, &resp); err != nil {
+		return "", err
+	}
+	return resp.Value, nil
+}
+
+func SimpleKVTest(net *oasis.Network, rtc client.RuntimeClient) error {
+	signer := newSignerWrapper(net.Entities()[1].Signer())
+
+	if err := kvInsert(rtc, signer, 0, "test_key", "test_value"); err != nil {
+		return err
+	}
+
+	// TODO: Wait for next epoch between all of these...
+
+	val, err := kvGet(rtc, "test_key")
+	if err != nil {
+		return err
+	}
+	if val != "test_value" {
+		return fmt.Errorf("fetched value does not match inserted value")
+	}
+
+	if err := kvRemove(rtc, signer, 1, "test_key"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RegisterScenarios registers all oasis-sdk end-to-end runtime tests.
+func RegisterScenarios() error {
```

Comment:
Consider moving this into a separate file as this will become messy when we have multiple scenarios.